src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -216,25 +242,27 @@ private UnboundLambda BindAnonymousFunction(CSharpSyntaxNode syntax, DiagnosticB
             Debug.Assert(syntax != null);
             Debug.Assert(syntax.IsAnonymousFunction());
 
-            var (refKinds, types, names, isAsync) = AnalyzeAnonymousFunction(syntax, diagnostics);
-            if (!types.IsDefault)
+            var lambda = AnalyzeAnonymousFunction(syntax, diagnostics);
+            var data = (PlainUnboundLambdaState)lambda.Data;
```

Comment:
>var data = (PlainUnboundLambdaState)lambda.Data; [](start = 12, length = 48)

It looks like the general pattern that we follow around ```UnboundLambda``` is that we do not access ```Data``` directly and do not make assumptions what specific implementation is used for ```Data```. We add required APIs to ```UnboundLambda``` instead, which passes the request to ```Data``` and each implementation handles the request as appropriate. Consider following the same pattern here.

Reply:
That makes sense. Thanks

---
In reply to: [340358132](https://github.com/dotnet/roslyn/pull/38786#discussion_r340358132) [](ancestors = 340358132)